docs/developer-guides/migration-guide-1.4.0.md

```diff
@@ -0,0 +1,169 @@
+```
+curl -H -s 'content-type: application/json' <ogmios-host>:<ogmios-port> -d '{"jsonrpc": "2.0", "method":"queryLedgerState/utxo", "params":{"addresses":["<governance authority address>"]}}' | jq
+```
+Save the value of this UTXO. It will be burned in the next step and will be used to identify the new Partner Chain
+in all future smart contracts operations.
+3. Set up initial governance by running `init-governance` command using the smart contracts version 7.0.1:
+```
+./pc-contracts-cli init-governance --network testnet \
+-k payment.skey \
+-c <genesis utxo> \
+-g <governance authority public key>
+```
+This will create a transaction burning the genesis UTXO and establish the governance authority for the Partner Chain
+identified by this UTXO.
+4. Set the D-parameter by running `insert-d-parameter`:
+```
+./pc-contracts-cli insert-d-parameter \
+-k payment.skey \
+--network testnet \
+--genesis-utxo <genesis utxo> \
+--d-parameter-permissioned-candidates-count <P value> \
+--d-parameter-registered-candidates-count <R value>
+```
+5. Set the permissioned candidates:
+```
+./pc-contracts-cli update-permissioned-candidates \
+-k payment.skey
+--network testnet \
+--genesis-utxo <genesis utxo> \
+--add-candidate <main chain pubkey>:<aura pubkey>:<grandpa pubkey> \
+...
+```
+repeating the `--add-candidate` argument for each permissioned candidate.
+
+### Re-register candidates
+
+At this point all SPOs willing to continue participating in the chain must re-register themselves with the newly created
+Partner Chain on Cardano. To do this, each SPO needs to do the following:
+
+1. Pick a registration UTXO from their UTXOs.
+Again, these can be obtained from Ogmios:
+```
+curl -H 'content-type: application/json' <ogmios-host>:<ogmios-port> -d '{"jsonrpc": "2.0", "method":"queryLedgerState/utxo", "params":{"addresses":["<SPO address>"]}}' | jq
+```
+This UTXO must not be burned before the `register` command is run.
+2. Obtain regitration signatures by running the following command using the 1.4.0 version of the node:
+```
+./partner-chains-node-v1.4.0 registration-signatures \
+--registration-utxo <registration utxo> \
+--genesis-utxo <genesis utxo> \
+--mainchain-signing-key <main chain skey> \
+--sidechain-signing-key <aura skey> | jq .
+```
+3. Perform registration by running:
+```
+./pc-contracts-cli register \
+-k payment.skey \
+--ada-based-staking \
+--network testnet \
+--genesis-utxo <genesis utxo> \
+--registration-utxo <registration utxo> \
+--spo-public-key <main chain pubkey> \
+--spo-signature <main chain signature> \
+--sidechain-public-keys <sidechain pubkey>:<aura pubkey>:<grandpa pubkey> \
+--sidechain-signature <sidechain signature>
+```
+
+After this step the SPO should be ready to be included in post-migration committee selection.
+
+### Upgrade the runtime to 1.4.0
+
+This step will switch the Partner Chain from observing the old on-chain Cardano data to the new one created in the
+previous steps. 
```

Comment:
Critical missing information is that the governance authority should wait at least two epoch after receiving confirmation from SPOs that they have registered